garden_ai/notebook_metadata.py

```diff
@@ -221,6 +177,126 @@ def save_requirements_data(
         return None
 
 
+def display_metadata_widget():
+    """
+    Displays the metadata editor widget
+    When one of the widgets fields is changed, pickles and saves the updated NotebookMetadata.
+    When the notebook is saved, the post_save_hook in custom_jupyter_config will
+    go and look for the pickled NotebookMetadata and save it to the notebooks metadata.
+    """
+
```

Comment:
This looks good! It really cleans up the top of the notebook and stays out of the way when you don't need it, and is easy to use when updating the metadata.